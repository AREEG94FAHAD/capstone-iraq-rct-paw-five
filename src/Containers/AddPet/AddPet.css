/* page header */

.header__text {
    padding: 30px 80px;
}
.add-pet-header {
    color: #e79559;
    font-weight: bold;
}

.add-pet-subheader {
    padding: 40px 20px;
    width: 80%;
    margin-bottom: 50px;
    color: #4b4b4b;
}

.header__photo img {
    width: 35rem;
    height: 35rem;
    margin-top: 30px;
    z-index: 1;
    position: absolute;
    z-index: 10;
    top: 5px;
    right: 0;
}

.addpet-form {
    background-color: #f2f2f2;
    padding: 30px 100px;
}
.addpet-form h2 {
    color: #e79559;
}
.error-message {
    color: red;
    font-size: 14px;
    margin-top: 8px;
}
.field-margin {
    margin-bottom: 60px;
}
.addpet-form input {
    background-color: #f2f2f2;
    border: none;
    border-bottom: 2px solid #e79559;
    border-radius: 0;
}
.addpet-row-space {
    margin: 0 0 0 90px;
}

.addpet-form label {
    padding: 0 30px 0 0px;
}

.custom-file-upload {
    border: 1px solid #e79559;
    border-radius: 30px;
    cursor: pointer;
}
.upload-image {
    display: block;
    margin: 0 auto;
    width: 8rem;
    height: 8rem;
}
.addpet-form button {
    background-color: #e79559;
    border: none;
    border-radius: 30px;
    width: 10rem;
}
.btn-primary:hover,
.btn-primary.active .btn-primary.focus {
    background: #f2726f;
    box-shadow: none;
    outline: none;
}
.form-control:focus {
    box-shadow: none;
    border-bottom: 3px #f2726f solid;
    background-color: #f2f2f2;
}

@media (max-width: 768px) {
    .add-pet-header {
        padding: 30px 10px;
    }
    .add-pet-subheader {
        padding: 1px 20px;
        width: auto;
        margin-bottom: 1px;
        color: #4b4b4b;
    }
    .addpet-form {
        padding: 30px 60px;
    }
    .addpet-form h2 {
        color: #e79559;
    }

    .header__text {
        padding: 30px 0;
        text-align: center;
    }
    .header__photo img {
        display: block;
        margin: 0 auto;
        width: 15rem;
        height: 15rem;
        margin-top: 1px;
        z-index: 1;
        position: static;
    }
    .addpet-row-space {
        margin: 0 0 0 0;
    }
}

@media (min-width: 769px) and (max-width: 992px) {
    .add-pet-subheader {
        padding: 1px 20px;
        width: auto;
        text-align: center;
        margin-top: 40px;
        margin-bottom: 1px;
        color: #4b4b4b;
    }
    .addpet-row-space {
        margin: 0 0 0 0;
    }
    .header__photo img {
        display: block;
        margin: 0 auto;
        width: 25rem;
        position: static;
    }
}

@media (min-width: 1024px) and (max-width: 1300px) {
    .header__photo img {
        width: 30rem;
        height: 30rem;
    }
}

@media (min-width: 992px) and (max-width: 1024px) {
    .header__photo img {
        width: 20rem;
        height: 30rem;
    }
}
