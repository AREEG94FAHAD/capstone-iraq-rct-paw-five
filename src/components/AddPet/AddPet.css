.form-borders {
  background-color: #f2f2f2 !important;
  border-bottom: 2px solid #e79559 !important;
}
.uplod-border {
  border: 1px solid #e79559 !important;
  cursor: pointer;
}

@media (min-width: 992px) {
  .cat-image {
    position: absolute;
    top: 0;
    right: 0;
    z-index: 10;
  }
}
@media (max-width: 992px) {
  .w-form {
    width: 100% !important;
  }
}
