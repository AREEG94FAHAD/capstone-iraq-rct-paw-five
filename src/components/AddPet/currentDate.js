const today = new Date()

const date = `${today.getFullYear()}-${
  today.getMonth() + 1
}-${today.getDate()} ${today.getHours()}:${today.getMinutes()}:${today.getSeconds()}`

export default date
