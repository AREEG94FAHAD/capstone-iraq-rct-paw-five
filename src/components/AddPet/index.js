export { default } from './AddPet'
export { default as AddPetForm } from './AddPetForm'
