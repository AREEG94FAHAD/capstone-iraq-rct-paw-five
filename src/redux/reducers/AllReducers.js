import { combineReducers } from 'redux'
import petReducer, {
  langReducer,
  myPets,
  deleteMyPet,
  updateMyPet,
} from './petReducer'
import user from './user'

export default combineReducers({
  langReducer,
  pets: petReducer,
  user,
  myPets,
  deleteMyPet,
  updateMyPet,
})
