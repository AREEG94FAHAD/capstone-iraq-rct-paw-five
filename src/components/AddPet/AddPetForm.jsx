import React from 'react'
import { useTranslation } from 'react-i18next'
import { Form, Col, Button } from 'react-bootstrap'
import PropTypes from 'prop-types'
import './AddPet.css'

function AddPetForm({
  handleSubmit,
  setFieldValue,
  file,
  handleChange,
  handleBlur,
  values,
  errors,
  touched,
}) {
  const { t } = useTranslation()
  const Values = values
  return (
    <div className="py-1 px-5" style={{ backgroundColor: '#f2f2f2' }}>
      <div className="px-lg-5">
        <h2
          className="font-weight-bold text-center text-lg-left mb-5 mt-4"
          style={{ color: '#e79559' }}
        >
          {t('add-pet.pets-info')}
        </h2>
        <Form onSubmit={handleSubmit}>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="petName"
                onBlur={handleBlur}
                value={values.petName}
                placeholder={t('add-pet.pet-name')}
                className={`${
                  touched.petName && errors.petName ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.petName && errors.petName ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.petName}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl className="a">
              <Form.Control
                onChange={handleChange}
                name="age"
                onBlur={handleBlur}
                value={values.age}
                placeholder={t('add-pet.pet-age')}
                className={`${
                  touched.age && errors.age ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.age && errors.age ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.age}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
          </Form.Row>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="species"
                onBlur={handleBlur}
                value={values.species}
                placeholder={t('add-pet.pet-species')}
                className={`${
                  touched.species && errors.species ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.species && errors.species ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.species}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl className="">
              <Form.Control
                onChange={handleChange}
                name="weight"
                onBlur={handleBlur}
                value={values.weight}
                placeholder={t('add-pet.pet-weight')}
                className={`${
                  touched.weight && errors.weight ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.weight && errors.weight ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.weight}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
          </Form.Row>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="color"
                onBlur={handleBlur}
                value={values.color}
                placeholder={t('add-pet.pet-color')}
                className={`${
                  touched.color && errors.color ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.color && errors.color ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.color}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl>
              <p>{t('add-pet.gender')}</p>
              <div className="d-flex justify-content-start mb-4">
                <Form.Check
                  className="mr-3"
                  onChange={() => {
                    Values.gender = 'male'
                  }}
                  type="radio"
                  onBlur={handleBlur}
                  name="gender"
                  value="Male"
                />

                <Form.Label className="mr-3">{t('add-pet.male')}</Form.Label>
                <Form.Check
                  className="mr-3"
                  onChange={() => {
                    Values.gender = 'female'
                  }}
                  type="radio"
                  name="gender"
                  value="Female"
                />
                <Form.Label className="mr-3">{t('add-pet.female')}</Form.Label>
              </div>
              <div className="mb-5"> </div>
            </Col>
          </Form.Row>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="story"
                onBlur={handleBlur}
                value={values.story}
                placeholder={t('add-pet.pet-story')}
                className={`${
                  touched.story && errors.story ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.story && errors.story ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.story}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl className="">
              <div>
                <Form.Label
                  htmlFor="file"
                  className=" text-center d-flex flex-column align-items-center justify-content-center bg-transparent border rounded uplod-border rounded-3 w-75 w-form mt-3"
                >
                  <p className="d-block text-center my-auto py-2">
                    {t('add-pet.upload-photo')}
                  </p>
                </Form.Label>
              </div>
              <input
                id="file"
                name="file"
                type="file"
                accept="images/*"
                onChange={e => {
                  setFieldValue('file', e.currentTarget.files[0])
                }}
                className="d-none"
              />

              {file && (
                <img
                  src={file}
                  className="d-block mt-4 mx-auto"
                  alt="your pet"
                  width="200"
                  height="200"
                />
              )}
            </Col>
          </Form.Row>
          <h2
            className="font-weight-bold text-center text-lg-left mb-5 mt-5"
            style={{ color: '#e79559' }}
          >
            {t('add-pet.contact-info')}
          </h2>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="ownerName"
                onBlur={handleBlur}
                value={values.ownerName}
                placeholder={t('add-pet.ownerName')}
                className={`${
                  touched.ownerName && errors.ownerName ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.ownerName && errors.ownerName ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.ownerName}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl className="addpet-row-space">
              <Form.Control
                onChange={handleChange}
                name="aboutOwner"
                onBlur={handleBlur}
                value={values.aboutOwner}
                placeholder={t('add-pet.aboutOwner')}
                className=" mb-5 form-borders rounded-0 border-0 w-75 w-form"
              />
            </Col>
          </Form.Row>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="email"
                type="email"
                onBlur={handleBlur}
                value={values.email}
                placeholder={t('add-pet.email')}
                className={`${
                  touched.email && errors.email ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.email && errors.email ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.email}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="phoneNumber"
                onBlur={handleBlur}
                value={values.phoneNumber}
                placeholder={t('add-pet.phoneNumber')}
                className={`${
                  touched.phoneNumber && errors.phoneNumber ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.phoneNumber && errors.phoneNumber ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.phoneNumber}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
          </Form.Row>
          <Form.Row>
            <Col lg xl>
              <Form.Control
                onChange={handleChange}
                name="city"
                onBlur={handleBlur}
                value={values.city}
                placeholder={t('add-pet.city')}
                className={`${
                  touched.city && errors.city ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.city && errors.city ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.city}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
            <Col lg xl className="addpet-row-space">
              <Form.Control
                onChange={handleChange}
                name="address"
                onBlur={handleBlur}
                value={values.address}
                placeholder={t('add-pet.address')}
                className={`${
                  touched.address && errors.address ? 'mb-2' : 'mb-5'
                } form-borders rounded-0 border-0 w-75 w-form`}
              />
              {touched.address && errors.address ? (
                <div className="mb-2 text-center text-lg-left text-danger">
                  {errors.address}
                </div>
              ) : (
                <div className="mb-5"> </div>
              )}
            </Col>
          </Form.Row>
          <div className="d-flex d-flex justify-content-center mr-lg-5">
            <Button
              type="submit"
              className="btn px-5 mr-lg-5 my-4 text-dark border-0"
              style={{ backgroundColor: '#e79559' }}
            >
              {t('add-pet.submit')}
            </Button>
          </div>
        </Form>
      </div>
    </div>
  )
}
AddPetForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  setFieldValue: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  file: PropTypes.string.isRequired,
  values: PropTypes.arrayOf(PropTypes.object).isRequired,
  errors: PropTypes.arrayOf(PropTypes.object).isRequired,
  touched: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default AddPetForm
