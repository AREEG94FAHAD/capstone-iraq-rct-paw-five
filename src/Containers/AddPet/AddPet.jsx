import React, { useState } from 'react'
import './AddPet.css'
import { useFormik } from 'formik'
import * as Yup from 'yup'
import { useTranslation } from 'react-i18next'
import { Form, Col, Button } from 'react-bootstrap'
import firebase, { storage } from '../../firebase'
import AddPetForm from './AddPetForm'
import 'firebase/firestore'
import Cat from '../../images/pets-vectors/addPetCat.svg'
import Paw from '../../images/icons/paw.svg'

function AddPet() {
    const { t } = useTranslation()
    const [file, setFile] = useState(null)

    const validationSchema = Yup.object().shape({
        petName: Yup.string().required(t('add-pet.required')),
        species: Yup.string().required(t('add-pet.required')),
        color: Yup.string().required(t('add-pet.required')),
        story: Yup.string().required(t('add-pet.required')),
        ownerName: Yup.string().required(t('add-pet.required')),
        email: Yup.string()
            .required(t('add-pet.required'))
            .email(t('add-pet.email-validation')),
        phoneNumber: Yup.string(t('add-pet.example-number'))
            .required(t('add-pet.required'))
            .matches(/^[0-9]{11}$/, t('add-pet.phone-validation')),
        city: Yup.string().required(t('add-pet.required')),
        address: Yup.string().required(t('add-pet.required')),
    })
    const {
        handleSubmit,
        setFieldValue,
        handleChange,
        handleBlur,
        resetForm,
        values,
        errors,
        touched,
    } = useFormik({
        initialValues: {
            petName: '',
            age: '',
            species: '',
            weight: '',
            color: '',
            gender: '',
            story: '',
            file: null,
            ownerName: '',
            aboutOwner: '',
            email: '',
            phoneNumber: '',
            city: '',
            address: '',
        },
        validationSchema,
        onSubmit() {
            const uploadTask = storage
                .ref(`petImages/${values.file.name}`)
                .put(values.file)
            uploadTask.on(
                'state_changed',

                () => {
                    storage
                        .ref('petImages')
                        .child(values.file.name)
                        .getDownloadURL()
                        .then((link) => {
                            firebase
                                .firestore()
                                .collection('pets')
                                .add({
                                    ...values,
                                    file: link,
                                })
                                .then(() => resetForm())
                        })
                }
            )
        },
    })

    const reader = new FileReader()

    reader.onloadend = () => {
        setFile(reader.result)
    }
    if (values.file != null) {
        reader.readAsDataURL(values.file)
    }
    return (
        <div>
            {/* page header */}
            <div className="d-flex flex-column flex-lg-row justify-content-lg-between">
                <div className="header__text">
                    <div className="d-flex flex-column-reverse align-items-center flex-md-row justify-content-md-center justify-content-lg-start mt-4 mb-4">
                        <img src={Paw} alt="paw" className="mt-4 mt-md-0 " />
                        <h1 className="add-pet-header">
                            {t('add-pet.add for adoption')}
                        </h1>
                    </div>
                    <div className="add-pet-subheader">
                        <h4>{t('add-pet.make-pets-happy')}</h4>
                    </div>
                </div>
                <div className="header__photo">
                    <img src={Cat} alt="cat" />
                </div>
            </div>
            <AddPetForm
                handleSubmit={handleSubmit}
                setFieldValue={setFieldValue}
                handleChange={handleChange}
                handleBlur={handleBlur}
                values={values}
                errors={errors}
                touched={touched}
                file={file}
            />
        </div>
    )
}

export default AddPet
