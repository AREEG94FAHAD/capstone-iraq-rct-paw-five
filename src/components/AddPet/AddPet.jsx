import React, { useState } from 'react'
import { useSelector } from 'react-redux'
import { useFormik } from 'formik'
import * as Yup from 'yup'
import { useTranslation } from 'react-i18next'
import { Row, Col } from 'react-bootstrap'
import { Redirect } from 'react-router-dom'
import firebase, { storage } from '../../firebase'
import AddPetForm from './AddPetForm'
import 'firebase/firestore'
import Cat from '../../images/Add-pet/AddPetCat.svg'
import Paw from '../../images/Add-pet/paw.svg'
import date from './currentDate'


function AddPet() {

  const { t } = useTranslation()
  const [file, setFile] = useState(null)
  const userState = useSelector(state => state.user)



  const validationSchema = Yup.object().shape({
    petName: Yup.string().required(t('add-pet.required')),
    species: Yup.string().required(t('add-pet.required')),
    age: Yup.string(t('add-pet.must-number')).max(
      2,
      t('add-pet.age-validation')
    ),
    weight: Yup.string(t('add-pet.weight-validation')).max(
      3,
      t('add-pet.weight-validation')
    ),
    color: Yup.string().required(t('add-pet.required')),
    story: Yup.string().required(t('add-pet.required')),
    ownerName: Yup.string().required(t('add-pet.required')),
    email: Yup.string()
      .required(t('add-pet.required'))
      .email(t('add-pet.email-validation')),
    phoneNumber: Yup.string(t('add-pet.example-number'))
      .required(t('add-pet.required'))
      .matches(/^[0-9]{11}$/, t('add-pet.phone-validation')),
    city: Yup.string().required(t('add-pet.required')),
    address: Yup.string().required(t('add-pet.required')),
  })
  const {
    handleSubmit,
    setFieldValue,
    handleChange,
    handleBlur,
    resetForm,
    values,
    errors,
    touched,
  } = useFormik({
    initialValues: {
      petName: '',
      age: '',
      species: '',
      weight: '',
      color: '',
      gender: '',
      story: '',
      file: null,
      ownerName: '',
      aboutOwner: '',
      email: '',
      phoneNumber: '',
      city: '',
      address: '',
      // date:currentDate,
      adopted: false
    },
    validationSchema,
    onSubmit() {
      const uploadTask = storage
        .ref(`petImages/${values.file.name}`)
        .put(values.file)
      uploadTask.on(
        'state_changed',

        () => {
          storage
            .ref('petImages')
            .child(values.file.name)
            .getDownloadURL()
            .then(link => {
              if(userState.user){
              firebase
                .firestore()
                .collection('pets')
                .add({
                  ...values,
                  adopted: false,
                  currentDate:date,
                  uid:userState.user.uid,
                  file: link,
                })
                .then(() => resetForm())
              }
            })
        }
      )
    },
  })

  const reader = new FileReader()

  reader.onloadend = () => {
    setFile(reader.result)
  }
  if (values.file != null) {
    reader.readAsDataURL(values.file)
  }
  return !userState ? (
    <Redirect to="/login" />
  ) : (
    <div className="overflow-hidden">
      {/* page header */}
      <Row className="mb-4">
        <Col className="pt-5 pl-lg-5 ml-lg-4">
          <div className="d-flex flex-column-reverse align-items-center flex-md-row justify-content-md-center justify-content-lg-start mt-4 mb-4">
            <img
              src={Paw}
              alt="paw"
              className="mt-4 mt-md-0 mr-lg-2 "
              width="55"
              height="55"
            />
            <h1
              className="font-weight-bold fs-1 text-center text-md-left px-2"
              style={{ color: '#e79559' }}
            >
              {t('add-pet.add for adoption')}
            </h1>
          </div>
          <div className="text-muted w-75 w-form ml-lg-5 pl-lg-2 mb-4">
            <h4 className="px-3 text-center text-lg-left">
              {t('add-pet.make-pets-happy')}
            </h4>
          </div>
        </Col>
        <Col lg={4} className="d-flex justify-content-end ml-lg-5">
          <img src={Cat} alt="cat" className="w-100 cat-image" height="450" />
        </Col>
      </Row>
      <AddPetForm
        handleSubmit={handleSubmit}
        setFieldValue={setFieldValue}
        handleChange={handleChange}
        handleBlur={handleBlur}
        values={values}
        errors={errors}
        touched={touched}
        file={file}
      />
    </div>
  )
}

export default AddPet
