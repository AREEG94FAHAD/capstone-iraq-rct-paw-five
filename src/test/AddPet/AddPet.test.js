import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import AddPet from '../../components/AddPet'
import AllReducers from '../../redux/reducers/AllReducers'

it('renders correctly', () => {
  const store = createStore(AllReducers)

  const tree = renderer
    .create(
      <Provider store={store}>
        <AddPet />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
