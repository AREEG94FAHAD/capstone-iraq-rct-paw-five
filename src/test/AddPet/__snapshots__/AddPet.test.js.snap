// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="d-flex flex-column flex-lg-row justify-content-lg-between"
  >
    <div
      className="header__text"
    >
      <div
        className="d-flex flex-column-reverse align-items-center flex-md-row justify-content-md-center justify-content-lg-start mt-4 mb-4"
      >
        <img
          alt="paw"
          className="mt-4 mt-md-0 "
          src="paw.svg"
        />
        <h1
          className="add-pet-header"
        >
          add-pet.add for adoption
        </h1>
      </div>
      <div
        className="add-pet-subheader"
      >
        <h4>
          add-pet.make-pets-happy
        </h4>
      </div>
    </div>
    <div
      className="header__photo"
    >
      <img
        alt="cat"
        src="addPetCat.svg"
      />
    </div>
  </div>
  <div
    className="addpet-form"
  >
    <h2
      className="font-weight-bold text-center text-lg-left mb-5 mt-4"
    >
      add-pet.pets-info
    </h2>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="petName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-name"
            value=""
          />
          <div
            className="field-margin error-message"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <input
            className="form-control"
            name="age"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-age"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="species"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-species"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <input
            className="form-control"
            name="weight"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-weight"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="color"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-color"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <p>
            add-pet.gender
          </p>
          <div
            className="d-flex justify-content-start"
          >
            <div
              className="mr-1 form-check"
            >
              <input
                className="form-check-input position-static"
                disabled={false}
                name="gender"
                onBlur={[Function]}
                onChange={[Function]}
                type="radio"
                value="Male"
              />
            </div>
            <label
              className="form-label"
            >
              add-pet.male
            </label>
            <div
              className="addpet-radio form-check"
            >
              <input
                className="form-check-input position-static"
                disabled={false}
                name="gender"
                onChange={[Function]}
                type="radio"
                value="Female"
              />
            </div>
            <label
              className="form-label"
            >
              add-pet.female
            </label>
          </div>
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="story"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.pet-story"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <div>
            <label
              className=" text-center d-flex flex-column align-items-center justify-content-center bg-transparent custom-file-upload form-label"
              htmlFor="file"
            >
              <p
                className="d-block text-center my-auto py-2"
              >
                add-pet.upload-photo
              </p>
            </label>
          </div>
          <input
            accept="images/*"
            className="d-none"
            id="file"
            name="file"
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
      <h2
        className="font-weight-bold text-center text-lg-left mb-5 mt-5"
      >
        add-pet.contact-info
      </h2>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="ownerName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.ownerName"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <input
            className="form-control"
            name="aboutOwner"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.aboutOwner"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.email"
            type="email"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <input
            className="form-control"
            name="phoneNumber"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.phoneNumber"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-xl col-lg"
        >
          <input
            className="form-control"
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.city"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
        <div
          className="addpet-row-space col-xl col-lg"
        >
          <input
            className="form-control"
            name="address"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="add-pet.address"
            value=""
          />
          <div
            className="field-margin"
          >
             
          </div>
        </div>
      </div>
      <div
        className="d-flex d-flex justify-content-center"
      >
        <button
          className="btn btn btn-primary"
          disabled={false}
          type="submit"
        >
          add-pet.submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
